src/dialog-holder.component.ts

```diff
@@ -29,84 +29,102 @@ export class DialogHolderComponent {
+    const componentRef = this.viewContainer.createComponent(factory, options.index);
+    const dialogWrapper: DialogWrapperComponent = <DialogWrapperComponent> componentRef.instance;
+    const _component: DialogComponent<T, T1> =  dialogWrapper.addComponent(component);
+
+    if (typeof(options.index) !== 'undefined') {
       this.dialogs.splice(options.index, 0, _component);
-    }
-    else {
+    } else {
       this.dialogs.push(_component);
     }
-    setTimeout(()=>{
-      dialogWrapper.container.nativeElement.classList.add('show');
-      dialogWrapper.container.nativeElement.classList.add('in');
+    setTimeout(() => {
+      dialogWrapper.wrapper.nativeElement.classList.add('show');
+      dialogWrapper.wrapper.nativeElement.classList.add('in');
     });
-    if(options.autoCloseTimeout) {
-      setTimeout(()=>{
+    if (options.autoCloseTimeout) {
+      setTimeout(() => {
         this.removeDialog(_component);
       }, options.autoCloseTimeout);
     }
-    if(options.closeByClickingOutside) {
-      dialogWrapper.closeByClickOutside();
+    if (options.closeByClickingOutside) {
+      dialogWrapper.onClickOutsideModalContent( () => {
+        this.removeDialog(dialogWrapper.content);
+      });
     }
-    if(options.backdropColor) {
-      dialogWrapper.container.nativeElement.style.backgroundColor = options.backdropColor;
+    if (options.backdropColor) {
+      dialogWrapper.wrapper.nativeElement.style.backgroundColor = options.backdropColor;
     }
-    
-    this.updateBodyClass();
+    // to avoid circular references hand the dialog a callback for it to self close
+    _component.onClose(this.removeDialog.bind(this));
+
+    // update the body class depending on the count
+    this.toggleModalOpenClassOnBody();
 
     return _component.fillData(data);
   }
 
   /**
-   * Removes dialog
+   * Initiates the removal of a dialog from the collection,
+   * removal is deferred by 300ms to give classList updates enough
+   * to take effect
    * @param {DialogComponent} component
    */
-  removeDialog(component:DialogComponent<any, any>) {
-    let element = component.wrapper.container.nativeElement;
-
-    element.classList.remove('show');
-    element.classList.remove('in');
+  removeDialog(component: DialogComponent<any, any>) {
+    const containerEl = component.wrapper.nativeElement;
+    containerEl.classList.remove('show');
+    containerEl.classList.remove('in');
     setTimeout(() => {
         this._removeElement(component);
     }, 300);
   }
 
+
+  /**
+   * Insructs the holder element to remove all dialogs,
+   * and flushes the collections
+   */
+  removeAllDialogs() {
+    this.viewContainer.clear();
+    this.dialogs = [];
+  }
+
   /**
    * Bind a body class 'modal-open' to a condition of dialogs in pool > 0
    */
-  
-  private updateBodyClass() {
+
+  private toggleModalOpenClassOnBody() {
     const bodyClass = 'modal-open';
```

Comment:
Same for naming convetion here. To use `dialog-open`.

Reply:
Venturing here into the eventual 100% seperation from bootstrap which in one way is good, it might be worth adding a default styles sheet to the component then which plucks the styles we like from bootstrap. 

If we move this way - i'll need to rename the package lol. It will be a bit disingenuous to use Bootstrap's name if we're not Bootstrap first, native second.

Reply:
I Agree, but maybe it's better. I don't know the numbers, but maybe it's better to change the default name now that it's a new project than later, with more people using it.

Also, since there is nothing related to bootstrap, apart from classes, isn't it better to do this now and move along focusing on just the Angular?

I can easily provide a css file with the classes need. I already have it.

Reply:
True now better than later. Ok will make a note to create the new one and move things to it. Will need a migration guide , your css will be useful too.

Reply:
I'll make a PR with a css file using `.modal` syntax class names.